### PR TITLE
Run github actions with catkin

### PR DESCRIPTION
**Pull request by mgrrx**

### CODE REVIEW

**Review by jspricke — APPROVED** [review]

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -9,10 +9,18 @@ jobs:
...
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
+        with:
+          path: catkin_ws/src/aioros
       - run: apt update
       - name: Install dependencies
-        run: apt install -y python3-anyio python3-h11 python3-httpx python3-lxml python3-pytest python3-pytest-cov python3-pytest-mock python3-genpy python3-std-msgs python3-std-srvs python3-rosgraph-msgs python3-actionlib-msgs
-      - name: Run tests
-        run: PYTHONPATH=. pytest-3 -s --cov=aioros
+        run: apt install -y python3-anyio python3-h11 python3-httpx python3-lxml python3-pytest python3-pytest-cov python3-pytest-mock python3-genpy python3-std-msgs python3-std-srvs python3-rosgraph-msgs python3-actionlib-msgs catkin build-essential
```
Comment (jspricke) [inline .github/workflows/build.yaml:19]: maybe don't use `build-essential`?

**Comment by jspricke** [comment]
Btw. https://tracker.debian.org/pkg/ros-catkin-tools ;)

File: .github/workflows/build.yaml (hunk 1 above)
Comment (mgrrx) [reply .github/workflows/build.yaml:19]: what do you recommend instead?
I am still adding support for pytests in cmake…

Comment (jspricke) [reply .github/workflows/build.yaml:19]: g++, I guess

Comment (jspricke) [reply .github/workflows/build.yaml:19]: (I think you can teach cmake not to depend on it, not sure if it works for catkin)

Comment (mgrrx) [reply .github/workflows/build.yaml:19]: I'm creating this PR in preparation of the pybind tf2 wrapper, which means I will need a compiler soon ;)

Comment (jspricke) [reply .github/workflows/build.yaml:19]: ok. I think of build-essential as a Debian thing of the past which is hard to get rid of, not sure we should expose that to ROS, that's why I would propose to rather declare what you actually need.

Comment (mgrrx) [reply .github/workflows/build.yaml]: done

**Pull request merged**